NPE: getCurrentPosition

> Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'long android.media.MediaExtractor.getSampleTime()' on a null object reference
at org.antennapod.audio.SonicAudioPlayer.getCurrentPosition(SonicAudioPlayer.java:125)
at org.antennapod.audio.MediaPlayer.getCurrentPosition(MediaPlayer.java:690)

--- a/library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java
+++ b/library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java
@@ -120,6 +120,7 @@ public class SonicAudioPlayer extends AbstractAudioPlayer {
     public int getCurrentPosition() {
         switch (mCurrentState) {
             case STATE_ERROR:
+            case STATE_IDLE:
                 return 0;
             default:
                 return (int) (mExtractor.getSampleTime() / 1000);

--- a/library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java
+++ b/library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java
@@ -121,6 +121,7 @@ public class SonicAudioPlayer extends AbstractAudioPlayer {
         switch (mCurrentState) {
             case STATE_ERROR:
             case STATE_IDLE:
+            case STATE_INITIALIZED:
                 return 0;
             default:
                 return (int) (mExtractor.getSampleTime() / 1000);

